mex/extractors/international_projects/transform.py

```diff
@@ -135,10 +172,54 @@ def transform_international_projects_sources_to_extracted_activities(
     for source in international_projects_sources:
         if activity := transform_international_projects_source_to_extracted_activity(
             source,
+            international_projects_activity,
             extracted_primary_source,
             person_stable_target_ids_by_query_string,
             unit_stable_target_id_by_synonym,
             funding_sources_stable_target_id_by_query,
             partner_organizations_stable_target_id_by_query,
         ):
             yield activity
+
+
+def get_theme_for_activity_or_topic(
+    theme: list[dict[str, Any]],
+    activity1: str | None,
+    activity2: str | None,
+    topic1: str | None,
+    topic2: str | None,
+) -> list[Theme]:
+    """Get theme identifier for activities and topics.
+
+    Args:
+        theme: theme extracted from mapping
+        activity1: activity 1
+        activity2: activity 1
+        topic1: topic 1
+        topic2: topic 2
+
+    Returns:
+        Sorted list of Theme
+    """
+    default_theme_from_mapping: Theme = theme[0]["mappingRules"][0]["setValues"][0]
+    themes_dict_from_mapping: dict[str, Theme] = {}
+    for theme_item in theme:
+        for rule in theme_item["mappingRules"]:
+            themes_dict_from_mapping.update(
+                dict.fromkeys(rule["forValues"], rule["setValues"][0])
+            )
+
+    def get_theme_or_default(key: str | None) -> Theme:
+        if key:
+            if theme := themes_dict_from_mapping.get(key):
+                return theme
+
+        return default_theme_from_mapping
+
+    theme_set = set()
+    theme_set.add(get_theme_or_default(activity1))
+    theme_set.add(get_theme_or_default(activity2))
+    theme_set.add(get_theme_or_default(topic1))
+    theme_set.add(get_theme_or_default(topic2))
+
+    return sorted(list(theme_set), key=lambda x: x.name)
```

Comment:
```suggestion
     themes: list[str] = []
     if topic1 in theme[0]["mappingRules"][0]["forValues"]:
          themes.append(theme[0]["mappingRules"][0]["setValues"])
```
Again something likes this for all five cases, would be much much simple of a logic imo

Reply:
Sadly theme extraction from mapping has multiple forValues and setValues, and a rule where there needs to be a default value as the data coming from source excel file can be optional or different or empty. if we go with this approach, there will be alot of if and elses and code repetition. 
Have a look at `international-projects` activity mapping -> `theme`.

Reply:
Maybe I don't understand the rule, but I see precisely 3 ifs here